blazeutils/spreadsheets.py

```diff
@@ -302,3 +362,90 @@ def cell_decimal(self):
     def next_row(self):
         self.rownum += 1
         self.colnum = 0
+
+
+class _OpenpyxlReader:
+    def __init__(self, openpyxl_wb, sheetnum=0):
+        _openpyxl_required()
+        self.book = openpyxl_wb
+        self.sheetnum = sheetnum
+        # Openpyxl uses 1-based indexing for cells
+        self.rownum = 1
+        self.colnum = 1
+        self.sheet = self.book[self.book.sheetnames[sheetnum]]
+
+    @classmethod
+    def from_xlwt(cls, xlwt_wb):
+        raise NotImplementedError('openpyxl cannot read xls format')
+
+    @classmethod
+    def from_xlsx(cls, openpyxl_wb):
+        # Included for backwards compatibility
+        return cls(openpyxl_wb)
+
+    def cell_value(self, is_date=False):
+        try:
+            return self.sheet.cell(self.rownum, self.colnum).value
+        finally:
+            self.colnum += 1
+
+    def cell_date(self):
+        # Openpyxl automatically converts to date or datetime
+        value = self.cell_value()
+        if isinstance(value, dt.datetime):
+            return value.date()
+        elif isinstance(value, dt.date):
+            return value
+        else:
+            raise TypeError(f'{value}: not a date')
+
+    def cell_datetime(self):
+        # Openpyxl automatically converts to date or datetime
+        value = self.cell_value()
+        if isinstance(value, dt.datetime):
+            return value
+        elif isinstance(value, dt.date):
+            return dt.datetime(value.year, value.month, value.day)
+        else:
+            raise ValueError(f'{value}: not a datetime')
+
+    def cell_decimal(self):
+        value = self.cell_value()
+        return Decimal(value)
+
+    def next_row(self):
+        self.rownum += 1
+        self.colnum = 1
+
+
+class Reader(object):
+    if openpyxl:
+        reader_cls = _OpenpyxlReader
+    else:
+        reader_cls = _XlrdReader
```

Comment:
Same concern here as above.  We would end up with an exception that implicitly recommends xlrd, the deprecated library.  Also no ability to choose library if both are installed.

Reply:
Come to think of it, we can probably just decide which implementation to use based on the type of the workbook or spreadsheet object passed to the "base" class.